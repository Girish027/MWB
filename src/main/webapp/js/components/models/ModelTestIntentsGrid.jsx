import React, { Component } from 'react';
import PropTypes from 'prop-types';
import _ from 'lodash';
import { Table } from '@tfs/ui-components';
import Constants from 'constants/Constants';

export default class ModelTestIntentsGrid extends Component {
  constructor(props) {
    super(props);
    this.props = props;
  }

  get columns() {
    const columns = [
      {
        Header: Constants.MODEL_TEST_INTENTS_TABLE.rank.header,
        id: Constants.MODEL_TEST_INTENTS_TABLE.rank.header,
        accessor: Constants.MODEL_TEST_INTENTS_TABLE.rank.id,
        maxWidth: 200,
        Cell: (row) => <div>{row.index + 1}</div>,
      },
      {
        Header: Constants.MODEL_TEST_INTENTS_TABLE.intent.header,
        accessor: Constants.MODEL_TEST_INTENTS_TABLE.intent.id,
      },
      {
        Header: Constants.MODEL_TEST_INTENTS_TABLE.score.header,
        accessor: Constants.MODEL_TEST_INTENTS_TABLE.score.id,
        Cell: ({ value: score }) => {
          if (!_.isNil(score)) {
            score = `${score.toFixed(4)}`;
          }
          return score;
        },
      },
    ];
    return columns;
  }

  render() {
    return (
      <Table
        id="ModelTestIntentsGrid"
        resizable
        data={this.props.data}
        columns={this.columns}
        showPagination={false}
        defaultSorted={[
          {
            id: Constants.MODEL_TEST_INTENTS_TABLE.rank.header,
            desc: false,
          },
        ]}
      />
    );
  }
}

ModelTestIntentsGrid.propTypes = {
  data: PropTypes.array.isRequired,
};
