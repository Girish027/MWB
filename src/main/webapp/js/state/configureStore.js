import { createStore, applyMiddleware, combineReducers } from 'redux';
import thunkMiddleware from 'redux-thunk';
import { createLogger } from 'redux-logger';
import { reducer as formReducer } from 'redux-form';
import { appReducer, defaultState as defaultAppState } from 'state/reducers/app_reducers';
import { headerReducer, defaultState as defaultHeaderState } from 'state/reducers/header_reducers';
import { projectsManagerReducer } from 'state/reducers/projectsManager_reducers';
import { consistencyReportReducer, defaultState as defaultConsistencyReportState } from 'state/reducers/consistencyReport_reducers';
import { Reducers as gridReducers } from 'react-redux-grid';
import config from 'config';
import { Map } from 'immutable';
import { projectListSidebarReducer, defaultState as defaultProjectListSidebarState } from './reducers/projectListSidebar_reducers';
import { taggingGuideReducer, defaultState as defaultTaggingGuideState } from './reducers/taggingGuide_reducers';
import { taggingGuideImportReducer } from './reducers/taggingGuideImport_reducers';
import { tagDatasetsReducer, defaultState as defaultTagDatasetsState } from './reducers/tag_datasets_reducers';
import { createDatasetDialogReducer } from './reducers/createDatasetDialog_reducers';
import { cellEditableReducer } from './reducers/cellEditable_reducers';
import { cellEditableManualTagSuggestReducer } from './reducers/cellEditableManualTagSuggestReducer_reducers';
import { reportsReducer } from './reducers/reports_reducers';
import { configReducer } from './reducers/config_reducers';
import { preferencesReducer } from './reducers/preferences_reducer';

let store;

if (config.env != 'test') {
  const transformObject = (obj) => {
    try {
      return JSON.parse(JSON.stringify(obj));
    } catch (err) {
      return obj;
    }
  };
  const loggerMiddleware = createLogger({
    stateTransformer: (state) => transformObject(state),
    actionTransformer: (action) => transformObject(action),
  });

  const rootReducer = combineReducers({
    app: appReducer,
    form: formReducer,
    ...gridReducers,
    config: configReducer,
    projectListSidebar: projectListSidebarReducer,
    header: headerReducer,
    taggingGuide: taggingGuideReducer,
    taggingGuideImport: taggingGuideImportReducer,
    projectsManager: projectsManagerReducer,
    tagDatasets: tagDatasetsReducer,
    consistencyReport: consistencyReportReducer,
    createDatasetDialog: createDatasetDialogReducer,
    cellEditable: cellEditableReducer,
    cellEditableManualTagSuggest: cellEditableManualTagSuggestReducer,
    reports: reportsReducer,
    preferences: preferencesReducer,
  });

  const configureStore = preloadedState => createStore(
    rootReducer,
    preloadedState,
    (config.env != 'production')
      ? applyMiddleware(thunkMiddleware, loggerMiddleware)
      : applyMiddleware(thunkMiddleware),
  );

  store = configureStore();
} else {
  /* store for tests */
  /* ToDo: move to separate file */

  let testProjects = Map();
  testProjects = testProjects.set('-1', {
    id: '-1',
    clientId: '-1',
    name: 'Test Project',
    created: 1498754327022,
    locale: 'en-US',
    description: 'Test Project Description',
    vertical: 'OTHER',
  });
  let testProjectDatasets = Map();
  const datasetsMap = Map({
    '-10': {
      _key: '-10',
      id: '-10',
      clientId: '-1',
      projectId: '-1',
      name: 'Dataset One',
      type: 'Audio/Voice (Live)',
      description: 'Test Dataset One',
      locale: 'en-US',
      created: 1503012662726,
      status: 'COMPLETED',
      task: 'INDEX',
      percentComplete: undefined,
      errorCode: undefined,
    },
    '-11': {
      _key: '-11',
      id: '-11',
      clientId: '-1',
      projectId: '-1',
      name: 'Dataset Two',
      type: 'Audio/Voice (Live)',
      description: 'Test Dataset Two',
      locale: 'en-US',
      created: 1503052662727,
      status: 'COMPLETED',
      task: 'INDEX',
      percentComplete: undefined,
      errorCode: undefined,
    },
    '-12': {
      _key: '-12',
      id: '-12',
      clientId: '-1',
      projectId: '-1',
      name: 'Dataset Three',
      type: 'Chat/Text',
      description: 'Test Dataset Three',
      locale: 'en-US',
      created: 1503032662727,
      status: 'NULL',
    },
    '-13': {
      _key: '-13',
      id: '-13',
      clientId: '-1',
      projectId: '-1',
      name: 'Dataset Four',
      type: 'Chat/Text',
      description: 'Test Dataset Four',
      locale: 'en-US',
      created: 1503031662727,
      status: 'COMPLETED',
      task: 'INDEX',
      percentComplete: undefined,
      errorCode: undefined,
    },
  });
  testProjectDatasets = testProjectDatasets.set('-1', datasetsMap);


  const mockStore = {
    app: {
      ...defaultAppState,
      userId: '-1',
      csrfToken: 'test',
    },
    header: {
      ...defaultHeaderState,
      client: { id: '-1', name: 'Test Client' },
      username: 'TestUser@247-inc.com',
    },
    consistencyReport: {
      ...defaultConsistencyReportState,
    },
    projectsManager: {
      projects: testProjects,
      projectDatasets: testProjectDatasets,
      trackDatasetsIds: {},
      projectModels: Map(),
      projectConfigs: Map(),
      projectIntents: Map(),
      trackModelsIds: {},
      trackModelsBatchTestIds: {},

    },
    projectListSidebar: {
      ...defaultProjectListSidebarState,
    },
    taggingGuide: {
      ...defaultTaggingGuideState,
    },
    tagDatasets: {
      ...defaultTagDatasetsState,
    },
    config: { },
    preferences: { },
  };

  const rootReducer = combineReducers({
    app: appReducer,
    header: headerReducer,
    consistencyReport: consistencyReportReducer,
    projectsManager: projectsManagerReducer,
    projectListSidebar: projectListSidebarReducer,
    taggingGuide: taggingGuideReducer,
    tagDatasets: tagDatasetsReducer,
    config: configReducer,
    preferences: preferencesReducer,
  });

  store = createStore(rootReducer, mockStore, applyMiddleware(thunkMiddleware));
}

export default store;
