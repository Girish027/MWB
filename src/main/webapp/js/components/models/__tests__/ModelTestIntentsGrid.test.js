import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import toJSON from 'enzyme-to-json';
import ModelTestIntentsGrid from 'components/models/ModelTestIntentsGrid';

const middlewares = [];
const mockStore = configureStore(middlewares);

// Initialize mockstore with empty state
const initialState = {
  header: {
    client: {
      id: 5,
    },
  },
};
const store = mockStore(initialState);

describe('ModelTestIntentsGrid', () => {
  let wrapper;
  describe('Creating an instance', () => {
    beforeAll(() => {
      const testProps = {
        datasets: {},
        project: {},
        data: [],
      };
      wrapper = shallow(<Provider store={store}>
        <ModelTestIntentsGrid {...testProps} />
      </Provider>);
    });
    test('should exist', () => {
      expect(wrapper.exists()).toBe(true);
    });
  });

  describe('Snapshots', () => {
    beforeAll(() => {
      const testProps = {
        datasets: {},
        project: {},
        data: [],
      };
      wrapper = shallow(<ModelTestIntentsGrid
        {...testProps}
      />);
    });
    test('renders correctly for ModelTestIntentsGrid table', () => {
      expect(toJSON(wrapper)).toMatchSnapshot();
    });
  });
});
