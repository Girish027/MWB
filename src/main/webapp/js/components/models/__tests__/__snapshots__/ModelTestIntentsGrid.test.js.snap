// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModelTestIntentsGrid Snapshots renders correctly for ModelTestIntentsGrid table 1`] = `
<withTheme(Table)
  columns={
    Array [
      Object {
        "Cell": [Function],
        "Header": "Rank",
        "accessor": "rank",
        "id": "Rank",
        "maxWidth": 200,
      },
      Object {
        "Header": "Intent",
        "accessor": "intent",
      },
      Object {
        "Cell": [Function],
        "Header": "Score",
        "accessor": "score",
      },
    ]
  }
  data={Array []}
  defaultSorted={
    Array [
      Object {
        "desc": false,
        "id": "Rank",
      },
    ]
  }
  id="ModelTestIntentsGrid"
  resizable={true}
  showPagination={false}
/>
`;
